scripts/enable_gitlab.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+set -eu
+set -o pipefail
+
+generate_random () {
+	dd if=/dev/urandom bs=1k count=1 2>/dev/null | sha256sum -b | cut -f1 -d ' '
+}
+
+enable_gitlab() {
+	local MYSQL_PASSWORD
+
+	mkdir -p /etc/protonet/gitlab
+	touch /etc/protonet/gitlab/enabled
+
+	if [ -f /etc/protonet/gitlab/mysql_passwd ]; then
+		MYSQL_PASSWORD=$(</etc/protonet/gitlab/mysql_passwd)
+	else
+		MYSQL_PASSWORD=$(generate_random)
+		echo -n $MYSQL_PASSWORD > /etc/protonet/gitlab/mysql_passwd
+	fi
+
+	MYSQL_QUERY="CREATE DATABASE IF NOT EXISTS gitlab;
+	CREATE USER IF NOT EXISTS 'gitlab'@'%';
+	SET PASSWORD FOR 'gitlab'@'%' = PASSWORD('$MYSQL_PASSWORD');
+	GRANT ALL PRIVILEGES ON gitlab.* TO 'gitlab'@'%';"
+	echo "$MYSQL_QUERY"
+	docker exec -i mysql mysql --password=s3kr3t --batch <<< "$MYSQL_QUERY"
+
+	sed -r "s/DB_PASS=[a-zA-Z0-9]+/DB_PASS=$MYSQL_PASSWORD/" -i /etc/systemd/system/gitlab.service
```

Comment:
This will break on the next update, right? Because the service unit would be overwritten. I'd rather suggest reading in the value stored in  /etc/protonet/gitlab/mysql_passwd into the service at launchtime